packages/core/src/node/cluster/master-process.ts

```diff
@@ -33,8 +40,11 @@ export class MasterProcess extends EventEmitter {
         if (!this.serverWorker) {
             throw new Error('Server worker is not running.');
         }
+        this.isRestarting = true;
         this.emit('restarting', this.serverWorker);
         console.log(`Restarting the server worker is requested.`);
+
+        await this.serverWorker.stop();
```

Comment:
An old worker should be stopped only in the case if a new worker is successfully started and initialized, please rollback.

Reply:
I don't really understand, because I made this PR so that we would try to shutdown processes gracefully on backend startup errors, such as `EADDRINUSE`. If we try to start a new worker while the old one is still running and still bound to `localhost:3000`, then the new worker would crash just by trying to start, right ?

Tell me if I missed something.

Reply:
`restart` is called when a new extension is installed to spawn a new server worker If a new server worker cannot be started or initialized with that extension then we should continue using the old. Only when a new worker is successfully started and initialized it is allowed to stop old.

Our issues does not have anything to do with restarting, only with an initial start. It would be nice to minimize scope of changes to avoid breaking something.